File: CSharp/FaculdadeVincit.DesignPatterns.StrategyPattern/SacolaDeCompras.cs
namespace FaculdadeVincit.DesignPatterns.StrategyPattern;

public class SacolaDeCompras
{
    public IPagamentoStrategy? Pagamento { get; set; }

    public void RealizarPagamento(double valor)
    {
        if (Pagamento == null)
        {
            Console.WriteLine("Método de Pagamento ainda não informado!");
            return;
        }

        Pagamento.Pagar(valor);
    }
}

File: Python/strategy_pattern/sacola_de_compras.py
from pagamento_strategy import PagamentoStrategy

class SacolaDeCompras:
    pagamento : PagamentoStrategy = None
    
    def realizar_pagamento(self, valor: float):
        if self.pagamento is None:
            print("Método de pagamento não definido")
        else:
            self.pagamento.pagar(valor)